src/nvim/screen.c

```diff
@@ -3338,6 +3346,16 @@ win_line (
             char_attr = hl_combine_attr(spell_attr, char_attr);
         }
 
+        if (has_bufhl && v > 0) {
+          bufhl_attr = bufhl_get_attr(&bufhl_info, (colnr_T)v);
+          if( bufhl_attr != 0) {
+            if (!attr_pri)
+              char_attr = hl_combine_attr(char_attr, bufhl_attr);
+            else
+              char_attr = hl_combine_attr(bufhl_attr, char_attr);
+          }
+        }
```

Comment:
Would be ideal to have only a single fucntion call here, something like `handle_has_bufhl`. This will help a lot when merging upstream changes.